x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/transforms/pivot/SchemaUtil.java

```diff
@@ -158,13 +161,13 @@ public static void deduceMappings(
         getSourceFieldMappings(
             client,
             headers,
-            sourceIndex,
-            sourceQuery,
+            sourceConfig,
             allFieldNames.values().stream().filter(Objects::nonNull).toArray(String[]::new),
-            runtimeMappings,
             ActionListener.wrap(
                 sourceMappings -> listener.onResponse(
                     resolveMappings(
+                        transformId,
+                        Boolean.FALSE.equals(settingsConfig.getDeduceMappings()) == false,
```

Comment:
Nonblocking: could this be `Boolean.TRUE.equals(settingsConfig.getDeduceMappings())`? Or is there a reason we want to see if the value parses to False then check if that parsed value is not false? "does this value not parse to false" vs "does this value parse to true"?

Reply:
Yes, there is a difference.
Before we introduced `deduce_mappings` setting, mapping deduction was used for all transforms by default.
We wanted `true` to be the default if the user does not specify this setting (in order to maintain backwards compatibility).
So if `settingsConfig.getDeduceMappings()` is `null` (because the user didn't set it), the expression:
```
Boolean.FALSE.equals(settingsConfig.getDeduceMappings()) == false
```
evaulates to `true` which we want.

With your proposition, we would translate `null` to `false` which is not the default we want.